.github/workflows/goclean.yml

```diff
@@ -0,0 +1,38 @@
+name: goclean
+
+on:
+  push:
+    paths:
+      - '**.go'
+      - 'go.mod'
+      - 'go.sum'
+    branches: [ main ]
+  pull_request:
+    paths:
+      - '**.go'
+      - 'go.mod'
+      - 'go.sum'
+    branches: [ main ]
+  schedule:
```

Comment:
Shouldn't `on.push` make this irrelevant?

Reply:
I pondered that too, and I **think** so? The one corner case I can think of is that a change in the go version may cause new warnings to materialize, and it's best to catch those up front rather than in an arbitrary unrelated PR.

But perhaps that corner case is best handled with a simple comment in the file reminding anyone who revs the `go-version` to ensure `vet` and `fmt` are clean prior to commit?

WDYT?

Reply:
I didn't consider that. I think it's "cheap" enough to run the workflow on a schedule. If it becomes an issue for whatever reason, we can always disable it.

Reply:
SGTM